modules/op-changing-source-branch-in-repository-crd.adoc

```diff
@@ -0,0 +1,25 @@
+// This module is included in the following assembly:
+//
+// *cicd/pipelines/using-pipelines-as-code.adoc
+
+:_content-type: REFERENCE
+[id="changing-source-branch-in-repository-crd_{context}"]
+= Changing the source branch for the pipeline definition in the `Repository` CRD
+
+[role="_abstract"]
+By default, when processing a `push` or `pull request` event, {pac} fetches the pipeline definition from the branch that triggered the event. You can use the `pipelinerun_provenance` setting in the `Repository` CRD to fetch the definition from the default branch configured on the Git repository provider, such as `main`, `master`, or `trunk`.
+
+[source,yaml]
+----
+...
+spec:
+  ...
+  settings:
+    pipelinerun_provenance: "default_branch"
+  ...
+----
+
+[NOTE]
+====
+If your repository allows unverified contributors to create a pull request that triggers the CI/CD process, use this setting as a security precaution, so that {pac} does not run a pipeline definition from am unverified contributor.
```

Comment:
that's not quite accurate, it would not run pipeline definition from non repository contributor by default... but having the definition coming from the main branch mean there is no mistake allowing a malicious change in one of the pull request iteration without being properly vetted by someone who has the power to merge pull request on a repository.....

the folks at rhtap did some rephrasing as well in here if you want to take a look https://github.com/redhat-appstudio/docs.appstudio.io/pull/105/files#diff-ba65a1054bd3efadb9de7c75aaea43d64c615b107904fedc959562a08bea7a08R71